### PR TITLE
Implement a tomcat upgrade workaround for upgrade pipelines

### PR DESCRIPTION
**Pull request by ehelms**
See https://community.theforeman.org/t/performing-dnf-update-on-foreman-server-results-in-errors/34959/10

### CODE REVIEW

File: pipelines/upgrade/05-server_to_intermediate.yml [hunk 1]
```diff
@@ -19,6 +19,8 @@
       scenario: "{{ pipeline_type }}"
       scenario_os: "{{ pipeline_os }}"
       scenario_version: "{{ forklift_upgrade_version_intermediate }}"
+    - role: update_tomcat
+      when: pipeline_type == 'katello'
```
Comment (evgeni) [inline pipelines/upgrade/05-server_to_intermediate.yml:23]: and pipeline_os centos8-stream? Otherwise this will get executed on Alma and fail?

Comment (ehelms) [reply pipelines/upgrade/05-server_to_intermediate.yml]: Good point, done.

**Review by evgeni — APPROVED** [review]

**Review by ekohl — COMMENTED** [review]
Is this something users should also do?

**Comment by ehelms** [comment]
> Is this something users should also do?

They will have to to upgrade, see https://community.theforeman.org/t/performing-dnf-update-on-foreman-server-results-in-errors/34959/10

**Comment by ekohl** [comment]
Looks like this broke the pipelines:
```
[2023-09-11T08:05:33.956Z] fatal: [pipe-up-foreman-nightly-debian11]: FAILED! => 
[2023-09-11T08:05:33.956Z]   msg: |-
[2023-09-11T08:05:33.956Z]     The conditional check '(pipeline_type == 'katello' || pipeline_type == 'luna') && pipeline_os == 'centos8-stream'' failed. The error was: template error while templating string: expected token 'name', got '|'. String: {% if (pipeline_type == 'katello' || pipeline_type == 'luna') && pipeline_os == 'centos8-stream' %} True {% else %} False {% endif %}. expected token 'name', got '|'
[2023-09-11T08:05:33.956Z]   
[2023-09-11T08:05:33.956Z]     The error appears to be in '/home/pipe-debian11-upgrade/forklift/roles/update_tomcat/tasks/main.yml': line 2, column 3, but may
[2023-09-11T08:05:33.956Z]     be elsewhere in the file depending on the exact syntax problem.
[2023-09-11T08:05:33.956Z]   
[2023-09-11T08:05:33.956Z]     The offending line appears to be:
[2023-09-11T08:05:33.956Z]   
[2023-09-11T08:05:33.956Z]     ---
[2023-09-11T08:05:33.956Z]     - name: Set module hotfixes on AppStream
[2023-09-11T08:05:33.956Z]       ^ here
```

**Pull request merged**